api/tag/janitor.go

```
@@ -7,7 +7,26 @@ import (
 
 func NewJanitor(tagStore Store, taskProvider provider.TaskProvider) *janitor.Janitor {
 	return janitor.NewJanitor("Tag", func() error {
-		// todo: delete all tags that don't exist in taskProvider
+		tasks, err := taskProvider.List()
+		if err != nil {
+			return err
+		}
+
+		tags, err := tagStore.SelectByType("task")
+		if err != nil {
+			return err
+		}
+
+		m := make(map[string]bool, len(tasks))
+		for _, task := range tasks {
+			m[task.TaskID] = true
+		}
+
+		for _, tag := range tags {
+			if !m[tag.EntityID] {
+				tagStore.Delete(tag.EntityType, tag.EntityID, tag.Key)
```

you are swallowing the error again